src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -166,11 +170,19 @@ export const NewProposal = ({ navigation, data, query }) => {
 
     if (newProposalData.documents) {
       let documents = JSON.parse(newProposalData.documents);
+      let documentsLength = documents.length;
+
       variables.attributes = {
         ...variables.attributes,
         documentsAttributes: []
       };
 
+      while (documentsLength--) {
+        if (documents[documentsLength].url) {
+          documents.splice(documentsLength, 1);
+        }
+      }
```

Comment:
what is this exactly for?

Reply:
If a proposal with a PDF inside is desired to be edited, these PDFs have an absolute url. If we do not delete these urls from the array, an error message will come from the server. I thought I needed a countdown while loop to prevent this.

Reply:
could the same result be achieved with a `filter` method rejecting the entries with `url`? i guess it would be more readable that way.

Reply:
[11cd880](https://github.com/ikuseiGmbH/smart-village-app-app/pull/539/commits/11cd880fd4a1818be68e2554dd6598e4211f65e7)